app/gui/view/component-browser/component-list-panel/grid/src/lib.rs

```diff
@@ -473,10 +473,15 @@ impl Model {
         }
     }
 
-    fn entry_to_select_after_reset(&self) -> Option<(Row, Col)> {
-        let sections = [SectionId::Popular, SectionId::SubModules, SectionId::LocalScope].iter();
+    fn entry_to_select_after_reset(&self, info: &content::Info) -> Option<(Row, Col)> {
+        let top_module_sections =
+            (0..info.namespace_section_count).into_iter().map(SectionId::Namespace);
+        let sections = iter::once(SectionId::Popular)
+            .chain(top_module_sections)
+            .chain(iter::once(SectionId::LocalScope))
+            .collect::<Vec<_>>();
```

Comment:
Why do we need to collect Vec here?

Reply:
My mistake, it was a quick change to get the compiler happy, and forgot to change it afterward. Removed now.